ROOT-HEAD fails with error: use of deleted function 'std::__ROOT::span<T>& std::__ROOT::span<T>::operator=(std::__ROOT::span<T>&&) [with T = double]'

### Check duplicate issues.

- [x] Checked for duplicates

### Description

We have an error in the nightlies build for dev3cuda-x86_64-el9-gcc11-opt and dev3cuda-x86_64-el8-gcc11-opt
Here, CUDA is enabled. The build without CUDA are fine.

The error seems to be related to the RooFit::Evaluator::assignToGPU function:
```
/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roofit/roofitcore/src/RooFit/Evaluator.cxx:574:68: error: use of deleted function 'std::__ROOT::span<T>& std::__ROOT::span<T>::operator=(std::__ROOT::span<T>&&) [with T = double]'
```
Also this throws an error
```
/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/roofit/roofitcore/src/RooFit/Evaluator.cxx:575:48: error: cannot convert 'std::__ROOT::span<double>' to 'const std::__ROOT::span<const double>&'
  575 |    _evalContextCUDA.set(node, _evalContextCUDA._currentOutput);
```

The full log for ROOT-HEAD can be found here: https://lcgapp-services.cern.ch/cdash/upload/405718ff1f150aae3de5d392f40f70fd238ea254/ROOT-HEAD-build.log

### Reproducer

/

### ROOT version

master

### Installation method

lcgcmake from source: options [gitlab.cern.ch/sft/lcgcmake/-/blob/master/projects/CMakeLists.txt?ref_type=heads#L82](https://gitlab.cern.ch/sft/lcgcmake/-/blob/master/projects/CMakeLists.txt?ref_type=heads#L82)
### Operating system

alma8 alma9

### Additional context

_No response_

--- a/roofit/roofitcore/src/RooFit/Evaluator.cxx
+++ b/roofit/roofitcore/src/RooFit/Evaluator.cxx
@@ -571,7 +571,7 @@ void Evaluator::assignToGPU(NodeInfo &info)
                                              : _bufferManager->makeGpuBuffer(nOut);
       buffer = info.buffer->gpuWritePtr();
    }
-   _evalContextCUDA._currentOutput = std::span<double>{buffer, nOut};
+   assignSpan(_evalContextCUDA._currentOutput, {buffer, nOut});
    _evalContextCUDA.set(node, _evalContextCUDA._currentOutput);
    node->doEval(_evalContextCUDA);
    CudaInterface::cudaEventRecord(*info.event, *info.stream);

